includes/acl/compression/impl/normalize_streams.h

```diff
@@ -100,28 +100,114 @@ namespace acl
 			{
+
+
+#ifdef ACL_PRECISION_BOOST
+
+				// In our compressed format, we store the center value of the track range quantized on 8 bits.
+				// To get the best accuracy, we pick the values closest to the true minimum that is slightly lower, and true maximum that is slightly higher.
+				// This is to ensure that we encompass the lowest and highest values even after quantization.
+				const rtm::vector4f range_min = range.get_min();
+				const rtm::vector4f quantized_min0 = rtm::vector_clamp(rtm::vector_add(rtm::vector_floor(rtm::vector_mul_add(range_min, max_range_value, half)), half_range_value), zero, max_range_value);
+
+#else
+
 				// In our compressed format, we store the minimum value of the track range quantized on 8 bits.
 				// To get the best accuracy, we pick the value closest to the true minimum that is slightly lower.
 				// This is to ensure that we encompass the lowest value even after quantization.
 				const rtm::vector4f range_min = range.get_min();
 				const rtm::vector4f scaled_min = rtm::vector_mul(range_min, max_range_value);
 				const rtm::vector4f quantized_min0 = rtm::vector_clamp(rtm::vector_floor(scaled_min), zero, max_range_value);
+
+#endif
+
 				const rtm::vector4f quantized_min1 = rtm::vector_max(rtm::vector_sub(quantized_min0, one), zero);
 
+#ifdef ACL_PRECISION_BOOST
+
+				const rtm::vector4f padded_range_min0 = rtm::vector_mul_add(quantized_min0, inv_max_range_value, half_neg);
+
+#else
+
 				const rtm::vector4f padded_range_min0 = rtm::vector_mul(quantized_min0, inv_max_range_value);
 				const rtm::vector4f padded_range_min1 = rtm::vector_mul(quantized_min1, inv_max_range_value);
 
+#endif
+
 				// Check if min0 is below or equal to our original range minimum value, if it is, it is good
 				// enough to use otherwise min1 is guaranteed to be lower.
 				const rtm::mask4f is_min0_lower_mask = rtm::vector_less_equal(padded_range_min0, range_min);
+
+#ifdef ACL_PRECISION_BOOST
+
+				rtm::vector4f quantized_min = rtm::vector_select(is_min0_lower_mask, quantized_min0, quantized_min1);
+
+				const rtm::vector4f range_max = range.get_max();
+				const rtm::vector4f quantized_max0 = rtm::vector_clamp(rtm::vector_add(rtm::vector_floor(rtm::vector_mul_add(range_max, max_range_value, half)), half_range_value), zero, max_range_value);
+				const rtm::vector4f quantized_max1 = rtm::vector_min(rtm::vector_add(quantized_max0, one), max_range_value);
+				const rtm::vector4f padded_range_max0 = rtm::vector_mul_add(quantized_max0, inv_max_range_value, half_neg);
+				const rtm::mask4f is_max0_higher_mask = rtm::vector_greater_equal(padded_range_max0, range_max);
+				rtm::vector4f quantized_max = rtm::vector_select(is_max0_higher_mask, quantized_max0, quantized_max1);
+
+				// Finally, we pad the range further so the midpoint has minimal quantization error.
+				const rtm::vector4f quantized_extent = rtm::vector_sub(quantized_max, quantized_min);
+				union MaskVector
```

Comment:
I'll create a task for myself to add and/or/xor support to masks in RTM so we can clean this up later